tests/SonarScanner.Shim.Tests/PathHelperTests.cs

```diff
@@ -0,0 +1,181 @@
+        {
+            // Arrange
+            var path = "SomeDirectory\\";
+
+            // Act
+            var result = PathHelper.WithTrailingDirectorySeparator(new DirectoryInfo(path));
+
+            // Assert
+            result.Should().EndWith("\\");
+        }
+
+        [TestMethod]
+        public void WithTrailingDirectorySeparator_WhenDirectoryEndsWithSlash_ReturnsSameString()
+        {
+            // Arrange
+            var path = "SomeDirectory/";
+
+            // Act
+            var result = PathHelper.WithTrailingDirectorySeparator(new DirectoryInfo(path));
+
+            // Assert
+            result.Should().EndWith("\\");
+        }
+
+        [TestMethod]
+        public void WithTrailingDirectorySeparator_WhenDirectoryDoesNotEndWithSeparatorAndContainsDirectorySeparatorChar_ReturnsStringWithRightEnd()
+        {
+            // Arrange
+            var path = "SomeDirectory" + Path.DirectorySeparatorChar + "Foo";
+
+            // Act
+            var result = PathHelper.WithTrailingDirectorySeparator(new DirectoryInfo(path));
+
+            // Assert
+            result.Should().EndWith(Path.DirectorySeparatorChar.ToString());
+        }
+
+        [TestMethod]
+        public void WithTrailingDirectorySeparator_WhenDirectoryDoesNotEndWithSeparatorAndContainsAltDirectorySeparatorChar_ReturnsStringWithRightEnd()
+        {
+            // Arrange
+            var path = "SomeDirectory" + Path.AltDirectorySeparatorChar + "Foo";
+
+            // Act
+            var result = PathHelper.WithTrailingDirectorySeparator(new DirectoryInfo(path));
+
+            // Assert
+            result.Should().EndWith(Path.DirectorySeparatorChar.ToString());
+        }
+
+        [TestMethod]
+        public void WithTrailingDirectorySeparator_WhenDirectoryDoesNotEndWithSeparatorAndContainsNoSeparator_ReturnsStringWithDirectorySeparatorChar()
+        {
+            // Arrange
+            var path = "SomeDirectory";
+
+            // Act
+            var result = PathHelper.WithTrailingDirectorySeparator(new DirectoryInfo(path));
+
+            // Assert
+            result.Should().EndWith(Path.DirectorySeparatorChar.ToString());
+        }
+
+        [TestMethod]
+        public void IsInDirectory_ReturnsTheExpectedResults()
+        {
+            PathHelper.IsInDirectory(new FileInfo("C:\\Src\\File.cs"), new DirectoryInfo("C:\\Src\\")).Should().BeTrue();
+            PathHelper.IsInDirectory(new FileInfo("C:\\Src\\File.cs"), new DirectoryInfo("C:\\Src")).Should().BeTrue();
+            PathHelper.IsInDirectory(new FileInfo("C:\\Src\\File.cs"), new DirectoryInfo("C:\\")).Should().BeTrue();
+
+            PathHelper.IsInDirectory(new FileInfo("C:\\SrcFile.cs"), new DirectoryInfo("C:\\Src")).Should().BeFalse();
+            PathHelper.IsInDirectory(new FileInfo("C:\\Src\\File.cs"), new DirectoryInfo("C:\\Src\\Bar")).Should().BeFalse();
```

Comment:
Missing test cases for case-insensitivity

Reply:
Good point!